### PR TITLE
Update command system to 1.12.

### PR DESCRIPTION
**Pull request by JBYoshi**
Fixes SpongePowered#1390.

### CODE REVIEW

File: src/main/java/org/spongepowered/common/command/SpongeProxySource.java [hunk 1]
```diff
@@ -35,133 +36,151 @@
...
     }
 
     @Override
-    public String getIdentifier() {
-        return getOriginalSource().getIdentifier();
+    public MessageChannel getMessageChannel() {
+        return this.channel;
+    }
+
+    @Override public void setMessageChannel(MessageChannel channel) {
```
Comment (stephan-gh) [inline src/main/java/org/spongepowered/common/command/SpongeProxySource.java]: Add a new line after the `@Override` please :)

File: src/main/java/org/spongepowered/common/mixin/core/command/MixinFunctionObject.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import java.util.Map;
+
+@Mixin(FunctionObject.class)
+public abstract class MixinFunctionObject {
+
+    @Redirect(method = "create", at = @At(value = "INVOKE", target = "Ljava/util/Map;containsKey(Ljava/lang/Object;)Z"))
```
Comment (stephan-gh) [inline src/main/java/org/spongepowered/common/mixin/core/command/MixinFunctionObject.java]: You need `remap = false` on the `@At` annotation to make sure Mixin doesn't complain when compiling

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: src/main/java/org/spongepowered/common/command/SpongeProxySource.java (hunk 1 above)
Comment (JBYoshi) [reply src/main/java/org/spongepowered/common/command/SpongeProxySource.java]: Fixed.

File: src/main/java/org/spongepowered/common/mixin/core/command/MixinFunctionObject.java (hunk 2 above)
Comment (JBYoshi) [reply src/main/java/org/spongepowered/common/mixin/core/command/MixinFunctionObject.java]: Fixed.

**Pull request merged**